### PR TITLE
Introducing the Re-Connect Promotional Surface

### LINKED ISSUE #3278: Deprecate Key Sharing - Add re-connect promotional surface
```[tasklist]
### Tasks
- [ ] https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3322
```

### PR DESCRIPTION
**Pull request by wjrosa**
Fixes #3278

## Changes proposed in this Pull Request:

This PR introduces a new Promotional Surface (a banner at the top of the Stripe extension settings page). This new piece will encourage merchants to re-connect their Stripe account using the new OAuth flow (by clicking the "Configure connection" button).

It is not dismissible. It shows up when the merchant has not reconnected yet and replaces the existing banner promoting the new checkout experience. When the merchant reconnects but is still using the old checkout experience, the original piece is shown.

## Testing instructions

- Checkout to this branch on your local environment (`add/reconnect-banner`)
- Run `npm install`, `npm run build:webpack` and `npm run up`
- Force the `is_connected_via_oauth` method return to `false` (`includes/connect/class-wc-stripe-connect.php#213`)
- Access the Stripe extension settings page: `wp-admin/admin.php?page=wc-settings&tab=checkout&section=stripe&panel=settings`
- Confirm that you can see the new promotional surface:
Screenshot 2024-07-30 at 15 19 14
- Force `is_connected_via_oauth` to `false`
- Confirm that the new promotional surface is gone

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)
-   [ ] Added the [needs docs label](https://github.com/woocommerce/woocommerce-gateway-stripe/labels?q=docs) (or does not apply)
-   [ ] Included this PR in the Release Thread scope (or does not apply)

### CODE REVIEW

File: client/settings/payment-settings/promotional-banner-section.js [hunk 1]
```diff
@@ -99,57 +96,117 @@ const PromotionalBannerSection = ( {
...
-					</CardColumn>
-				</CardInner>
-				<ButtonsRow>
-					<MainCTALink
-						variant="secondary"
-						data-testid="enable-the-new-checkout"
-						onClick={ handleButtonClick }
-					>
+	const handleReConnectButtonClick = () => {
+		jQuery( '#btn-configure-connection' ).trigger( 'click' );
```
Comment (Mayisha) [inline client/settings/payment-settings/promotional-banner-section.js]: What are your thoughts on handling the button click by sending necessary props from `client/settings/payment-settings/index.js` to this component? We should avoid mixing React and jQuery unless this is the only possible solution.

**Review by Mayisha — CHANGES_REQUESTED** [review]
Thanks @wjrosa for implementing this. I have tested the PR and it works as described.

- When my site has not been re-connected, I see the re-auth promotional banner.

Screenshot 2024-07-31 at 12 25 46 PM

- The account connection modal is displayed when I click the `Configure connection` button.


Screenshot 2024-07-31 at 12 26 10 PM

I have a feedback about avoiding JQuery with React. 

I also have a minor suggestion. This is not a blocker to this PR but I think it could be a good enhancement. We are showing the promotional banner on the `Settings` tab of the Stripe settings page. When a merchant comes to the Stripe settings page, they first land on the `Payment methods` tab and unless they go to the `Settings` tab, they won't see the banner or know about this. I think it would be more effective if we display the banner on both of the tabs.

File: client/settings/payment-settings/promotional-banner-section.js (hunk 1 above)
Comment (wjrosa) [reply client/settings/payment-settings/promotional-banner-section.js]: Oh ok! I just replaced it and used the props instead 👍

**Comment by diegocurbelo** [comment]
Hey @wjrosa!

How about changing the button to start the OAuth flow (by just using the oauthURL we use in the modal) without the need to open the modal... this would also make showing the banner in both the `Payment methods` and the `Settings` tabs easier.

**Comment by wjrosa** [comment]
Thanks for the suggestions, @diegocurbelo ! I made the changes now. I sent you additional context over Slack, let me know if you want me to revert the last commit.

Also, with this change, it is not possible to test the connection flow over localhost. Stripe returns an empty URL for both live and test in this case.

**Comment by diegocurbelo** [comment]
> Also, with this change, it is not possible to test the connection flow over localhost. Stripe returns an empty URL for both live and test in this case.

https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3337 fixes that

File: client/settings/payment-settings/promotional-banner-section.js [hunk 2]
```diff
@@ -99,57 +100,130 @@ const PromotionalBannerSection = ( {
...
-						data-testid="dismiss"
-					>
-						{ __( 'Dismiss', 'woocommerce-gateway-stripe' ) }
-					</DismissButton>
-				</ButtonsRow>
-			</CardBody>
+					</h4>
+					<p>
+						{ __(
+							'Re-authenticate your Stripe account using the new flow by clicking the "Configure connection" button and make your store safer.',
```
Comment (diegocurbelo) [inline client/settings/payment-settings/promotional-banner-section.js]:
```suggestion
							'Re-connect your Stripe account using the new authentication flow by clicking the "Re-authenticate" button and make your store safer.',
```

File: client/settings/payment-settings/promotional-banner-section.js [hunk 3]
```diff
@@ -99,57 +100,130 @@ const PromotionalBannerSection = ( {
...
+			<ButtonsRow>
+				<MainCTALink
+					variant="secondary"
+					data-testid="re-connect-checkout"
+					onClick={ handleReConnectButtonClick }
+				>
+					{ __(
+						'Configure connection',
+						'woocommerce-gateway-stripe'
+					) }
```
Comment (diegocurbelo) [inline client/settings/payment-settings/promotional-banner-section.js]:
```suggestion
					{ __( 'Re-authenticate', 'woocommerce-gateway-stripe' ) }
```

**Review by diegocurbelo — CHANGES_REQUESTED** [review]
Thanks for working on this @wjrosa!

The code looks good, and it tests well. 
I added a couple of suggested text changes.

**Pull request merged**